org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/trace/TraceRegionMergerTest.java

```diff
@@ -0,0 +1,381 @@
+/**
+ * Copyright (c) 2015, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.generator.trace;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Queue;
+import java.util.Set;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.xtext.xbase.lib.CollectionLiterals;
```

Comment:
Sidenote: For production code I'd prefer to avoid `CollectionLiterals`. For tests it's ok to me.

Reply:
i dont know a proper replacement for this

Reply:
will inline ugly code